theano/tensor/blas.py

```diff
@@ -319,7 +322,9 @@ def default_blas_ldflags():
     # readily available. We try to see if that's the case, rather
     # than disable blas. To test it correctly, we must load a program.
     # Otherwise, there could be problem in the LD_LIBRARY_PATH.
-    return try_blas_flag(['-lblas'])
+    ret = try_blas_flag(['-lblas'])
+    if ret:
+        return ' '.join(ret)
```

Comment:
It add a '-L' flag, which might be required for proper -lblas usage.  Since we don't test without it we don't know if it will work.

So I modified it to try without that first and only add it if it doesn't work.